app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,9 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!
+
   def index
```

Comment:
- [ ] as part of the requirements you should use CanCanCan for Post authorization in Post Controller , kindly fix it by adding : 

```ruby
  class PostsController < ApplicationController
    #Add Post authorization
    load_and_authorize_resource

    ...
  end
```